sw/host/opentitanlib/src/io/io_mapper.rs

```diff
@@ -0,0 +1,134 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+use super::{gpio, i2c, spi};
+use anyhow::Result;
+use std::collections::HashMap;
+use std::ops::{Deref, DerefMut};
+
+use crate::io::gpio::GpioError;
+use crate::transport::{TransportError, TransportInterfaceType};
+
+#[derive(Default, Clone)]
+pub struct AliasMapper {
+    map: HashMap<String, String>,
+}
+
+/// Given an pin/uart/spi/i2c port name, if the name is a known alias, return the underlying
+/// name/number, otherwise return the string as is.
+impl AliasMapper {
+    pub fn new() -> Self {
+        Self::default()
+    }
+
+    /// Return the `name` itself if it is not mapped.
+    pub fn resolve(&self, name: &str) -> String {
+        let name = name.to_uppercase();
+        match self.map.get(&name) {
+            Some(v) => {
+                if v.eq(&name) {
+                    name
+                } else {
+                    self.resolve(v)
+                }
+            }
+            None => name,
+        }
+    }
+
+    /// Return None if the name is not mapped.
+    pub fn try_resolve(&self, name: &str) -> Option<&String> {
+        if let Some(name) = self.map.get(name) {
+            return Some(name);
+        }
+        let name = name.to_uppercase();
+        self.map.get(&name)
+    }
+}
+
+impl Deref for AliasMapper {
+    type Target = HashMap<String, String>;
+
+    fn deref(&self) -> &Self::Target {
+        &self.map
+    }
+}
+
+impl DerefMut for AliasMapper {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.map
+    }
+}
+
+#[derive(Default)]
+pub struct IoMapper {
+    pub pin_map: AliasMapper,
+    pub uart_map: AliasMapper,
+    pub i2c_map: AliasMapper,
+    pub pin_conf_map: HashMap<String, gpio::PinConfiguration>,
+    pub spi_conf_map: HashMap<String, spi::SpiConfiguration>,
+    pub i2c_conf_map: HashMap<String, i2c::I2cConfiguration>,
+    pub strapping_conf_map: HashMap<String, HashMap<String, gpio::PinConfiguration>>,
```

Comment:
Do these need to be public?

Reply:
They are public because there's no constructor in this module and the `TransportWrapper` still need to access some fields. But I'm happy to change it in a follow-up PR.